### PR TITLE
Issue 1534: Fix zap scan parameters

### LINKED ISSUE #1534: As ACF tech lead, I need the nightly-owasp-scan CircleCI workflow to run successfully and exclude non-TDP URLs in scan report
### Context:

the [nightly OWASP scans](https://app.circleci.com/insights/github/raft-tech/TANF-app/workflows/nightly/overview?branch=raft-tdp-main&reporting-window=last-90-days) have been timing out more frequently, at the backend scan step. As a reminder, we run these security vulnerability scans and store them in the [staging backend](https://tdp-backend-staging.app.cloud.gov/admin/security/owaspzapscan/) (for now) to meet ACF OCIO compliance standards ([ref](https://github.com/raft-tech/TANF-app/issues/1161)). 

See screenshot of workflow 30-day success rate as of 2/15 below:

The error message in the job is: `Too long with no output (exceeded 10m0s): context deadline exceeded`. According to [CircleCI documentation](https://support.circleci.com/hc/en-us/articles/360045268074-Build-Fails-with-Too-long-with-no-output-exceeded-10m0s-context-deadline-exceeded-), 10 minutes is the default timeout setting, and some suggestions are offered therein to resolve, including increasing the timeout. 

After some investigations noted in the [2/14](https://github.com/raft-tech/TANF-app/issues/1534#issuecomment-1039208747) and  [2/15](https://github.com/raft-tech/TANF-app/issues/1534#issuecomment-1040352457) comments below, we decided that we need to **increase the timeout** and also **prevent the non-TDP URLs from being scanned for security vulnerabilities** which appears to be the root cause of the timeout. 

### ACs:
- [x] nightly-owasp-scan workflow scan runs without timing out at 10min
- [x] non-TDP URLs are not reported in the backend scan

### Tasks:
- [x] Decide strategy for excluding non-TDP URLs from being scanned
- [x] Increase timeout to > 10min
- [x] Determine strategy for testing fix for `qasp review`

**Supporting Documentation:**

- [backend scan report with non-TDP URLs](https://hhsgov.sharepoint.com/:b:/r/sites/TANFDataPortalOFA/Shared%20Documents/compliance/Zap/ZAP%20Scanning%20Report_21322.pdf?csf=1&web=1&e=dmynmw) and evidence of CI step (below)

- [ZAP issue re: mozilla URLs ](https://github.com/zaproxy/zaproxy/issues/6443)
- [syntax for increasing timeout](https://support.circleci.com/hc/en-us/articles/360045268074-Build-Fails-with-Too-long-with-no-output-exceeded-10m0s-context-deadline-exceeded-)

### PR DESCRIPTION
**Pull request by ADPennington**
## Summary of Changes
_Provide a brief summary of changes_

- updates zap version to latest available
- increases wait time for zap scan ci job to 60min
- updates url exclusion list s.t. only TDP URLs are scanned/reported. 

Pull request closes #1534 

## How to Test
_List the steps to test the PR_
- Check [here](https://app.circleci.com/pipelines/github/raft-tech/TANF-app?branch=feat%2F1534-updating-to-latest-zaproxy&filter=all) for the latest nightly backend scan (runs every hour at 10 after and 40 after until merge) and confirm that: (1) backend owasp scan did not timeout after 10 min and (2) only TDP sites are included in the results:
  - backend scan timing out after 10min and TDP sites included in the results (for successful jobs) (before)
1534_33p5
1534_33p3

  - backend scan **_not_** timing out after 10min and non-TDP sites excluded in the results  (after)
 1534_33p2

## Tabletop

- [ ] Ready for Tabletop (Reviewer check this)
- [ ] Tabletop scheduled (Assignee check this)
- [ ] Tabletop approval 

## Deliverables
_More details on how deliverables herein are assessed included [here](https://github.com/raft-tech/TANF-app/blob/raft-tdp-main/docs/How-We-Work/our-priorities-values-expectations.md#Deliverables)._

### [Deliverable 1: Accepted Features](https://github.com/raft-tech/TANF-app/blob/raft-tdp-main/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-1-Accepted-Features)

Checklist of ACs:
- [x] nightly-owasp-scan workflow scan runs without timing out at 10min
- [x] non-TDP URLs are not reported in the backend scan
+ [ ] **`lfrohlich`** and/or **`adpennington`**  confirmed that ACs are met.

### [Deliverable 2: Tested Code](https://github.com/raft-tech/TANF-app/blob/raft-tdp-main/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-2-Tested-Code)

+ Are all areas of code introduced in this PR meaningfully tested?
  + [ ] If this PR introduces backend code changes, are they meaningfully tested?
  + [ ] If this PR introduces frontend code changes, are they meaningfully tested?
+ Are code coverage minimums met?
  + [ ] Frontend coverage: [_insert coverage %_] (see `CodeCov Report` comment in PR)
  + [ ] Backend coverage: [_insert coverage %_] (see `CodeCov Report` comment in PR)

### [Deliverable 3: Properly Styled Code](https://github.com/raft-tech/TANF-app/blob/raft-tdp-main/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-3-Properly-Styled-Code)

+ [ ] Are backend code style checks passing on CircleCI?
+ [ ] Are frontend code style checks passing on CircleCI?
+ [ ] Are code maintainability principles being followed?

### [Deliverable 4: Accessible](https://github.com/raft-tech/TANF-app/blob/raft-tdp-main/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-4-Accessibility)

+ [ ] Does this PR complete the epic? 
+ [ ] Are links included to any other gov-approved PRs associated with epic?
+ [ ] Does PR include documentation for Raft's a11y review? 
+ [ ] Did automated and manual testing with `iamjolly` and `ttran-hub` using Accessibility Insights reveal any errors introduced in this PR?

_n/a: zap scan parameter changes only_

### [Deliverable 5: Deployed](https://github.com/raft-tech/TANF-app/blob/raft-tdp-main/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-5-Deployed)

+ [ ] Was the code successfully deployed via automated CircleCI process to development on Cloud.gov?

### [Deliverable 6: Documented](https://github.com/raft-tech/TANF-app/blob/raft-tdp-main/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-6-Code-documentation)

+ [ ] Does this PR provide background for why coding decisions were made?
+ [ ] If this PR introduces backend code, is that code easy to understand and sufficiently documented, both inline and overall?
+ [ ] If this PR introduces frontend code, is that code easy to understand and sufficiently documented, both inline and overall?
+ [ ] If this PR introduces dependencies, are their licenses documented?
+ [ ] Can reviewer explain and take ownership of these elements presented in this code review?

### [Deliverable 7: Secure](https://github.com/raft-tech/TANF-app/blob/raft-tdp-main/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-7-Secure)

+ [ ] Does the OWASP Scan pass on CircleCI?
+ [ ] Do manual code review and manual testing detect any new security issues?
+ [ ] If new issues detected, is investigation and/or remediation plan documented?

### CODE REVIEW

File: .circleci/config.yml
Comment (ADPennington) [inline .circleci/config.yml:90]: increased wait for owasp scan to 60min before a timeout occurs.

File: .circleci/config.yml
Comment (ADPennington) [inline .circleci/config.yml:585]: this is a temporary change to cron job frequency for testing. this runs every hour at 10 after and 40 after on `feat/1534-updating-to-latest-zaproxy` branch. this should be changed back to default before merge.

File: .circleci/config.yml
Comment (ADPennington) [inline .circleci/config.yml:589]: same as 🔼. should be changed back to `raft-tdp-main` before merge.

File: scripts/zap-scanner.sh
Comment (ADPennington) [inline scripts/zap-scanner.sh:61]: added 2 sets of URLs to global exclusions:
- URL that ends in `cdn.mozilla.net` (non-TDP site [ref](https://github.com/zaproxy/zaproxy/issues/6443#issuecomment-775226449))
- URL that ends in `amazonaws.com` (S3 buckets)

File: tdrs-backend/docker-compose.yml
Comment (ADPennington) [inline tdrs-backend/docker-compose.yml:6]: updated zap version to latest available. this was primarily to check for any fixes in how false positives are handled.

File: tdrs-backend/reports/zap.conf
Comment (ADPennington) [inline tdrs-backend/reports/zap.conf:124]: no change here. temporarily add non-TDP URLs to confirm that `outofscope` doesnt help with exclusions related to 100001.

File: scripts/zap-scanner.sh
Comment (abottoms-coder) [reply scripts/zap-scanner.sh:61]: So this is effective? We were able to resolve the issues?

Comment (ADPennington) [reply scripts/zap-scanner.sh:61]: yep ([example](https://17074-281707402-gh.circle-artifacts.com/0/tdrs-backend/reports/owasp_report.html)). still increased the timeout since this step in the workflow still takes a while.

**Review by abottoms-coder — CHANGES_REQUESTED** [review]
Local testing works well for me in addition to the artifacts within this PR as well. Will approve PR once the temporary changes are reverted.

**Review by abottoms-coder — APPROVED** [review]

**Comment by abottoms-coder** [comment]
Adding QASP label to run through those automated checks, if everything passes will go to merging.

**Pull request merged**
